decidim-core/app/commands/decidim/destroy_account.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This command destroys the user's account.
+  class DestroyAccount < Rectify::Command
+    # Destroy a user's account.
+    #
+    # user - The user to be updated.
+    # form - The form with the data.
+    def initialize(user, form)
+      @user = user
+      @form = form
+    end
+
+    def call
+      return broadcast(:invalid) unless @form.valid?
+
+      destroy_user_account!
+      broadcast(:ok)
+    end
+
+    private
+
+    def destroy_user_account!
+      @user.email = "deleted-user-#{SecureRandom.uuid}@example.org"
```

Comment:
`email` can be empty when the user is deleted. There's a special index for that (not sure about the model's validation) but I think is way better to deal it this way.